infra/examples-dev/gcp/google-workspace-variables.tf

```diff
@@ -25,3 +26,14 @@ variable "google_workspace_example_admin" {
   description = "user to impersonate for Google Workspace API calls (null for value of `google_workspace_example_user`)"
   default     = null # will failover to user
 }
+
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_google_workspace_gcp_project_id         = (var.google_workspace_gcp_project_id == null || var.google_workspace_gcp_project_id == "") && (contains(var.enabled_connectors, "gdirectory") || contains(var.enabled_connectors, "gcal") || contains(var.enabled_connectors, "gmail") || contains(var.enabled_connectors, "google-chat") || contains(var.enabled_connectors, "google-meet") || contains(var.enabled_connectors, "gdrive"))
+  validate_google_workspace_gcp_project_id_message = "The google_workspace_gcp_project_id var should be populated if a Google Workspace connector is enabled."
+  validate_google_workspace_gcp_project_id_check = regex(
```

Comment:
I don't really follow this kind of checks... matching regexp on message, I guess is to make it fail.
Probably this could work and is easier (to follow and understand, imho)

`validate_google_workspace_gcp_project_id_check = tobool(local.validate_google_workspace_gcp_project_id ? "true" : "The google_workspace_gcp_project_id var should be populated if a Google Workspace connector is enabled.")`

> Only boolean values, null, and the exact strings "true" and "false" can be converted to boolean. All other values will produce an error.

Reply:
yeah, this seems too hacky for me. It's hard to read the conditions, and doesn't really stop TF from working.

Reply:
Yes, the purpose is to make TF to fail if the condition is not meet:
![image](https://github.com/Worklytics/psoxy/assets/75276364/0934b727-80bd-49e3-879c-fddc99045dfa)